xiaomi.vacuum.gen1/firmwarebuilder/imagebuilder.sh

```diff
@@ -18,146 +18,200 @@
 # place english.pkg and v11_<fw_version>.pkg (e.g. v11_003077.pkg) in this folder
 # place your authorized_keys in this folder
 #
-IS_MAC=false
-FIRMWARE_VERSION=$1
-SOUNDFILE=english.pkg
 
-if [[ ! -n "$FIRMWARE_VERSION" ]]; then
-	echo "You need to specify a firmware version, e.g. 003094"
+if [[ $EUID -ne 0 ]]; then
+	echo "You must be a root user" 2>&1
 	exit 1
 fi
 
+IS_MAC=false
 if [[ $OSTYPE == darwin* ]]; then
 	# Mac OSX
 	IS_MAC=true
 	echo "Running on a Mac, adjusting commands accordingly"
 fi
 
-if [ ! -f /usr/bin/ccrypt ] && ! $IS_MAC; then
-    echo "Ccrypt not found! Please install it (e.g. by apt-get install ccrypt)"
+if [ ! -f /usr/bin/ccrypt -a "$IS_MAC" = false ]; then
```

Comment:
I don't know what -a does in this case? Is this a bash specific construct? The AND NOT was imo clearer for anyone without bash knowledge but this is probably the better way to do things in bash then? :)

Reply:
no need to spawn multiple [ instances to combine results, you can use -a (AND) inside single [ condition block

Reply:
Ok so this is specific for single condition blocks? when I want to add a third parameter I can again use `-a "$FOO" = true` ? In that case it would be better to read 👍